### PR TITLE
[Registry] Added service registry context

### PR DESCRIPTION
**Pull request by dantleech**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | #4814 
| License         | MIT

Added additional argument to the service registry indicating the context in
order to give more helpful exception messages.

So instead of:

```
Service of type "boolean" does not exist.
```

You get:

```
Grid field of type "boolean" does not exist.
```

The context is optional in order to preserve BC (I would personally consider making it mandatory).

### CODE REVIEW

File: src/Sylius/Bundle/ReportBundle/Resources/config/services.xml [hunk 1]
```diff
@@ -35,10 +35,12 @@
 
         <service id="sylius.registry.report.data_fetcher" class="%sylius.registry.report.class%">
             <argument>Sylius\Component\Report\DataFetcher\DataFetcherInterface</argument>
+            <argument>Data fetcher</argument>
         </service>
 
         <service id="sylius.registry.report.renderer" class="%sylius.registry.report.class%">
             <argument>Sylius\Component\Report\Renderer\RendererInterface</argument>
+            <argument>Renderer</argument>
```
Comment (dantleech) [inline src/Sylius/Bundle/ReportBundle/Resources/config/services.xml]: s/Renderer/Report renderer

File: src/Sylius/Bundle/SettingsBundle/Resolver/ResolverServiceRegistry.php [hunk 2]
```diff
@@ -30,7 +30,7 @@ class ResolverServiceRegistry extends ServiceRegistry
      */
     public function __construct($interface, SettingsResolverInterface $defaultResolver)
     {
-        parent::__construct($interface);
+        parent::__construct($interface, 'Settings resolver');
```
Comment (dantleech) [inline src/Sylius/Bundle/SettingsBundle/Resolver/ResolverServiceRegistry.php]: Wonder if this should be also defined in the DI configuration.

File: src/Sylius/Component/Registry/PrioritizedServiceRegistry.php [hunk 3]
```diff
@@ -32,12 +32,18 @@ class PrioritizedServiceRegistry implements PrioritizedServiceRegistryInterface
     protected $interface;
 
     /**
+     * @var string
+     */
+    protected $context;
+
+    /**
      * @param $interface
```
Comment (dantleech) [inline src/Sylius/Component/Registry/PrioritizedServiceRegistry.php]: Missing `@param`

File: src/Sylius/Bundle/SettingsBundle/Resolver/ResolverServiceRegistry.php (hunk 2 above)
Comment (adamelso) [inline src/Sylius/Bundle/SettingsBundle/Resolver/ResolverServiceRegistry.php]: It is possible to have more than one instance of this registry - so I would put it in the DI config. :+1:

**Comment by pjedrzejewski** [comment]
👍

**Comment by dantleech** [comment]
Whilst I am here, what do you think about modifying the `NotExistingServiceException` by passing the keys of the registered services, so the message becomes:

```
Grid field of type "bool" does not exist, existing grid field types: "boolean", "string", "number", "foobar"
```

**Comment by pjedrzejewski** [comment]
@dantleech Sounds great!

**Comment by dantleech** [comment]
Updated.

**Comment by pjedrzejewski** [comment]
Thanks @dantleech!

**Pull request merged**